### PR TITLE
Watch kafka config in consolidated channel controller

### LINKED ISSUE #318: Consolidated channel dispatcher is not getting the changes in config-kafka during runtime
**Describe the bug**
Consolidated channel dispatcher is not getting the changes in config-kafka during runtime

Main problem: channel dispatcher doesn't get the changes in `config-kafka`
- Create a consolidated channel
- Change stuff in `config-kafka`
- Changes are read by the controller (besides the problems in https://github.com/knative-sandbox/eventing-kafka/issues/393), but not reflected in the channel dispatcher

There might be the same story for KafkaSource.

 **Expected behavior**
Similar to distributed channel, watch the changes, and apply them.

2 alternatives for channel:
- Watch the configmap in the controller, recreate the dispatcher
- Watch the configmap in the dispatcher

Same story for KafkaSource.

 **To Reproduce**
- Create a consolidated channel
- Change stuff in `config-kafka`
- Changes are not reflected in the channel dispatcher

 **Knative release version**
0.20

 **Additional context**
Noticed during https://github.com/knative-sandbox/eventing-kafka/pull/305

### PR DESCRIPTION
**Pull request by aliok**
Fixes #318 

## Proposed Changes

- Step by step refactoring: move `config-kafka` watching related code into `/common` from `/channel/distributed/common`
- Start watching `config-kafka` in consolidated channel controller --> TBD in consolidated dispatcher as well

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

**Comment by travis-minke-sap** [comment]
Hey Ali - I took a quick pass through the PR and it looks fine to me.  I'd like to have @eric-sap take a look as well since he's worked on the config watching more recently.  He is out today though so if this is holding you up I can approve and he can post-review tomorrow ?

**Comment by eric-sap** [comment]
Looks good to me; all that refactoring seems okay; I did a quick manual smoke-test of the distributed channel using this branch and the watcher still appears to work.

**Comment by aliok** [comment]
hi @eric-sap @travis-minke-sap 

Thanks for your reviews.

I need to watch the configmap in dispatcher as well (and delete the configmap mounting logic). There will be more commits here.

**Comment by aliok** [comment]
I decided to proceed with this PR as is.

The consolidated dispatcher requires somewhat bigger refactoring and this PR would get super large.

Let's merge this first.

File: pkg/common/config/kafkaconfigmapwatcher.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+)
+
+// This function type is for a shim so that we can pass our own logger to the Observer function
+type LoggingObserver func(ctx context.Context, configMap *corev1.ConfigMap)
+
+//
+// Initialize The Specified Context With A ConfigMap Watcher
+// Much Of This Function Is Taken From The knative.dev sharedmain Package
+//
+func InitializeKafkaConfigMapWatcher(ctx context.Context, logger *zap.SugaredLogger, handler LoggingObserver, namespace string) error {
```
Comment (slinkydeveloper) [inline pkg/common/config/kafkaconfigmapwatcher.go:38]: Can you take the watcher as parameter here more than creating one internally? So you can reuse the watcher given to you by `NewController`

Comment (aliok) [reply pkg/common/config/kafkaconfigmapwatcher.go:38]: Yep, let me try that

Comment (aliok) [reply pkg/common/config/kafkaconfigmapwatcher.go]: Done, thanks. Let's see if it breaks the build :)

File: cmd/channel/distributed/receiver/main.go [hunk 2]
```diff
@@ -134,8 +135,13 @@ func main() {
 	// Create A New Stats StatsReporter
 	statsReporter := metrics.NewStatsReporter(logger)
 
+	// Create A Watcher On The Configuration Settings ConfigMap & Dynamically Update Configuration
+	// Since this is designed to be called by the main() function, the default KNative package behavior here
+	// is a fatal exit if the watch cannot be set up.
+	cmw := sharedmain.SetupConfigMapWatchOrDie(ctx, logger.Sugar())
```
Comment (slinkydeveloper) [inline cmd/channel/distributed/receiver/main.go:141]: You forgot to start the config map watcher, right? Or does this method start it?

File: pkg/common/config/kafkaconfigmapwatcher.go [hunk 3]
```diff
@@ -35,13 +35,7 @@ type LoggingObserver func(ctx context.Context, configMap *corev1.ConfigMap)
...
 // Much Of This Function Is Taken From The knative.dev sharedmain Package
 //
-func InitializeKafkaConfigMapWatcher(ctx context.Context, logger *zap.SugaredLogger, handler LoggingObserver, namespace string) error {
-
-	// Create A Watcher On The Configuration Settings ConfigMap & Dynamically Update Configuration
-	// Since this is designed to be called by the main() function, the default KNative package behavior here
-	// is a fatal exit if the watch cannot be set up.
-	watcher := sharedmain.SetupConfigMapWatchOrDie(ctx, logger)
-
+func InitializeKafkaConfigMapWatcher(ctx context.Context, watcher configmap.Watcher, logger *zap.SugaredLogger, handler LoggingObserver, namespace string) error {
```
Comment (slinkydeveloper) [inline pkg/common/config/kafkaconfigmapwatcher.go:38]: The start of the config map watcher is done by the caller (the owner of the watcher), not this method

**Review by slinkydeveloper — CHANGES_REQUESTED** [review]

File: pkg/common/config/kafkaconfigmapwatcher.go (hunk 3 above)
Comment (aliok) [reply pkg/common/config/kafkaconfigmapwatcher.go:38]: ok, gonna delete the `start` inside `InitializeKafkaConfigMapWatcher`.

thanks

**Review by slinkydeveloper — APPROVED** [review]

File: pkg/common/client/testing/util.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2020 The Knative Authors
```
Comment (matzew) [inline pkg/common/client/testing/util.go:2]: you should update your IDE template :-)

**Pull request merged**